File: Golang/lista_de_exercicios_01/classes/produto/produto.go
package produto

import "math"

type Produto struct {
	Nome  string
	Preco float64
}

func (p Produto) SomarProduto(p2 Produto) float64 {
	return roundFloat(p.Preco+p2.Preco, 2)
}

func roundFloat(val float64, precision uint) float64 {
	ratio := math.Pow(10, float64(precision))
	return math.Round(val*ratio) / ratio
}

File: Java/lista_de_exercicios_01/lista_de_exercicios_01/src/Classes/Produto.java
package Classes;

public class Produto {
    String nome;
    float preco;


    public Produto(String nome, float preco) {
        this.nome = nome;
        this.preco = preco;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public float getPreco() {
        return preco;
    }

    public void setPreco(float preco) {
        this.preco = preco;
    }

    public float somarProdutos(Produto a) {
        return Math.round((getPreco() + a.getPreco()) * 100.0f) / 100.0f;
    }
}
